data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/bulk/Es6BulkApiWrapper.java

```diff
@@ -0,0 +1,84 @@
+package org.opensearch.dataprepper.plugins.sink.opensearch.bulk;
+
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.client.opensearch._types.ErrorResponse;
+import org.opensearch.client.opensearch._types.OpenSearchException;
+import org.opensearch.client.opensearch.core.BulkRequest;
+import org.opensearch.client.opensearch.core.BulkResponse;
+import org.opensearch.client.transport.JsonEndpoint;
+import org.opensearch.client.transport.endpoints.SimpleEndpoint;
+import org.opensearch.client.util.ApiTypeHelper;
+
+import javax.ws.rs.HttpMethod;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class Es6BulkApiWrapper implements BulkApiWrapper {
+    static final String DUMMY_DEFAULT_INDEX = "dummy";
+    private final OpenSearchClient openSearchClient;
+
+    public Es6BulkApiWrapper(final OpenSearchClient openSearchClient) {
+        this.openSearchClient = openSearchClient;
+    }
+
+    @Override
+    public BulkResponse bulk(BulkRequest request) throws IOException, OpenSearchException {
+        final JsonEndpoint<BulkRequest, BulkResponse, ErrorResponse> endpoint = es6BulkEndpoint(request);
+        return openSearchClient._transport().performRequest(request, endpoint, openSearchClient._transportOptions());
+    }
+
+    private JsonEndpoint<BulkRequest, BulkResponse, ErrorResponse> es6BulkEndpoint(BulkRequest bulkRequest) {
+        return new SimpleEndpoint<>(
+                // Request method
+                request -> HttpMethod.POST,
+
+                // Request path
+                request -> {
+                    final String index = request.index() == null ? DUMMY_DEFAULT_INDEX : request.index();
```

Comment:
Should we have this condition? If the request doesn't have an index, perhaps we should fail. Otherwise, the data goes into an index named `dummy`.